kickoff/jsonexport.py

```diff
@@ -0,0 +1,244 @@
+    firefoxversions["LATEST_FIREFOX_DEVEL_VERSION"] = lastStable[0][0]
+    firefoxversions["LATEST_FIREFOX_RELEASED_DEVEL_VERSION"] = lastStable[0][0]
+    # esr
+    lastStable = getFilteredReleases("firefox", ["esr"], lastRelease=True)
+    firefoxversions["FIREFOX_ESR"] = lastStable[0][0]+"esr"
+    # esr next
+    lastStable = getFilteredReleases("firefox", ["esr"], lastRelease=True, ESR_NEXT=True)
+    if lastStable:
+        # If not found, that means that we are managing only a single ESR
+        firefoxversions["FIREFOX_ESR_NEXT"] = lastStable[0][0]+"esr"
+    # aurora
+    firefoxversions["FIREFOX_AURORA"] = config.AURORA_VERSION
+    # old version
+    firefoxversions["LATEST_FIREFOX_OLDER_VERSION"] = config.LATEST_FIREFOX_OLDER_VERSION
+    return jsonify(firefoxversions)
+
+
+def replaceValue(array, versionTag, version):
+    for locale, datalist in array.iteritems():
+        i = 0
+        for ver in datalist:
+            if ver == versionTag:
+                array[locale][i] = version
+            i += 1
+
+
+def generateLocateBuilds(buildsVersionLocales, l10nchangesets, lastVersion):
+    # The l10n format for desktop is 'locale changeset'
+    # parse it
+    for locale in l10nchangesets.split('\n'):
+        if "\t" in locale:
+            # Not sure why, but in some cases, \t is used a separator
+            localeCode = locale.split("\t")[0]
+        else:
+            localeCode = locale.split(" ")[0]
+        if localeCode not in buildsVersionLocales.keys():
+            buildsVersionLocales[localeCode] = [lastVersion]
+        else:
+            buildsVersionLocales[localeCode] += [lastVersion]
+
+    return buildsVersionLocales
+
+
+def fillAuroraVersion(buildsVersionLocales):
+    for localeCode in config.SUPPORTED_AURORA_LOCALES:
+        if localeCode not in buildsVersionLocales.keys():
+            buildsVersionLocales[localeCode] = [config.AURORA_VERSION]
+        else:
+            buildsVersionLocales[localeCode] += [config.AURORA_VERSION]
+
+    return buildsVersionLocales
+
+
+def updateBuilds(product):
+    buildsVersionLocales = {}
+    # Stable
+    lastStable = getFilteredReleases(product, ["major", "stability"], lastRelease=True, withL10N=True)
+    buildsVersionLocales = generateLocateBuilds(buildsVersionLocales, lastStable[0][2],
+                                                lastStable[0][0])
+
+    # beta
+    lastStable = getFilteredReleases(product, ["dev"], lastRelease=True, withL10N=True)
+    buildsVersionLocales = generateLocateBuilds(buildsVersionLocales, lastStable[0][2],
+                                                lastStable[0][0])
+
+    # esr
+    lastStable = getFilteredReleases(product, ["esr"], lastRelease=True, withL10N=True)
+    buildsVersionLocales = generateLocateBuilds(buildsVersionLocales, lastStable[0][2],
+                                                lastStable[0][0])
+    buildsVersionLocales = fillAuroraVersion(buildsVersionLocales)
```

Comment:
All this reassignment of buildsVersionLocales is confusing...this would be more readable if generateLocateBuilds and fillAuroraVersion returned new dicts, and you called .update() on buildsVersionLocales instead. Eg:
buildsVersionLocales = {}
buildsVersionLocales.update(generateLocateBuilds(lastStable[0][2], lastStable[0][0]))

I _think_ you'll end up with the same result since you're not using deeply nested dicts.